pkg/virtctl/vm/vm.go

```diff
@@ -389,9 +434,93 @@ func getDryRunOption(dryRun bool) []string {
 	return nil
 }
 
-func (o *Command) Run(args []string) error {
+func expandArgs() cobra.PositionalArgs {
+	return func(cmd *cobra.Command, args []string) error {
+		if filePath != "" && len(args) > 0 {
+			return fmt.Errorf("error file path and Virtual Machine name can not be both defined")
+		}
+
+		if filePath == "" && len(args) == 0 {
+			return fmt.Errorf("error invalid arguments - Virtual Machine name or file must be provided")
+		}
+
+		if outputFormat != YAML && outputFormat != JSON {
+			return fmt.Errorf("error not supported output format defined: %w", errors.New(outputFormat))
+		}
+
+		return nil
+	}
+}
+
+func readVMFromFile(filePath string) (*v1.VirtualMachine, error) {
+	vm := &v1.VirtualMachine{}
+
+	readFile, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading file %+w", err)
+	}
 
-	vmiName := args[0]
+	err = yaml.Unmarshal(readFile, vm)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding Virtual Machine %+w", err)
+	}
+
+	return vm, nil
+}
+
+func printExpandedVM(outputFormat string, expandedVm *v1.VirtualMachine) error {
+	var formatedOutput []byte
+	var err error
+
+	switch outputFormat {
+	case JSON:
+		formatedOutput, err = json.MarshalIndent(expandedVm, "", " ")
+	case YAML:
+		formatedOutput, err = yaml.Marshal(expandedVm)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	cmd.Print(string(formatedOutput))
+	return nil
+}
+
+func expandVirtualMachine(virtClient kubecli.KubevirtClient, namespace string, args []string) error {
+	var expandedVm *v1.VirtualMachine
+	var err error
+
+	if len(args) == 1 {
+		vmName := args[0]
+		expandedVm, err = virtClient.VirtualMachine(namespace).GetWithExpandedSpec(vmName)
+		if err != nil {
+			return fmt.Errorf("error expanding VirtualMachine %s: %w", vmName, err)
+		}
+	} else {
+		vm, err := readVMFromFile(filePath)
+		if err != nil {
+			return err
+		}
+
+		expandedVm, err = virtClient.ExpandSpec(vm.Namespace).ForVirtualMachine(vm)
```

Comment:
The namespace can be provided as an argument to the command right?

You're also not checking if this is even provided in the definition before attempting to use it.

I'd rather drop the use of any manifest provided namespace here and just require that one is provided with the command line when using `--file`.